pkg/config/config.go

```diff
@@ -521,18 +521,38 @@ func LoadWithoutSecret() error {
 	return load(Datadog, "datadog.yaml", false)
 }
 
+func checkForUnknownKeys(config Config) []string {
+	var unknownKeys []string
+	knownKeys := config.GetKnownKeys()
+	loadedKeys := config.AllKeys()
+	for _, key := range loadedKeys {
+		if _, found := knownKeys[key]; !found {
+			// Check if any subkey terminated with a '.*' wildcard is marked as known
+			// e.g.: apm_config.* would match all sub-keys of apm_config
+			splitPath := strings.Split(key, ".")
+			for j := range splitPath {
+				subKey := strings.Join(splitPath[:j+1], ".") + ".*"
+				if _, found = knownKeys[subKey]; found {
+					break
+				}
+			}
+			if !found {
+				unknownKeys = append(unknownKeys, key)
+			}
+		}
+	}
+	return unknownKeys
+}
+
 func load(config Config, origin string, loadSecret bool) error {
 	if err := config.ReadInConfig(); err != nil {
 		log.Warnf("Error loading config: %v", err)
 		return err
 	}
 
-	knownKeys := config.GetKnownKeys()
-	loadedKeys := config.AllKeys()
-	for _, key := range loadedKeys {
-		if _, found := knownKeys[key]; !found {
-			log.Warnf("Unknown key in config file: %v", key)
-		}
+	unknownKeys := checkForUnknownKeys(config)
```

Comment:
Super nit: this isn't actually _checking_ for unknown keys, it's returning them. So this should be called `getUnknownKeys` or `findUnknownKeys`.

Also, there is no point for the variable, you could just:
```go
for k := range findUnknownKeys(config) {
```

Reply:
👍 Changing it.

I find it more readable to declare the variable anyway, but I guess the compact version is more idiomatic to go.